tests/unit/src/unit/issues/Issue9736.hx

```diff
@@ -0,0 +1,12 @@
+package unit.issues;
+
+import haxe.Constraints.Function;
+
+class Issue9736 extends Test {
+  function test() {
+    ((null: Foo): Function);
+    utest.Assert.pass();
+  }
+}
+
+private abstract Foo(Function) from Function to Function {}
```

Comment:
The test would be better if it didn't have `from Function` so we can be sure that we actually test the `to Function` case in particular. Then again the direction here is really obvious, so we're quite unlikely to break this.